smolyay/basis.py

```diff
@@ -128,13 +134,12 @@ def __call__(self, x):
         Raises
         ------
         ValueError
-            if input is outside the domain [-1,1]
+            if input is outside the domain [-1, 1]
 
         """
-        if numpy.any(numpy.greater(x, 1)) or numpy.any(numpy.less(x, -1)):
-            raise ValueError("Input is outside the domain [-1,1]")
-
-        return scipy.special.eval_chebyt(self._n,x)
+        if self.outside_domain(x):
```

Comment:
This would be how you might use this:
```suggestion
        if not numpy.all(self.check_in_domain(x)):
```
or else
```py
        if not self.check_all_in_domain(x):
```